packages/client/src/workflow-client.ts

```diff
@@ -640,6 +696,128 @@ export class WorkflowClient extends BaseClient {
     return await decodeFromPayloadsAtIndex(this.dataConverter, 0, response.queryResult?.payloads);
   }
 
+  /**
+   * Uses given input to make an UpdateWorkflowExecution call to the service, returning a handle to the update.
+   *
+   * Used as the final function of the startUpdate interceptor chain
+   */
+  protected async _startUpdateHandler<Ret>(input: WorkflowUpdateInput): Promise<WorkflowUpdateHandle<Ret>> {
+    return this._startUpdate(
+      input,
+      temporal.api.enums.v1.UpdateWorkflowExecutionLifecycleStage.UPDATE_WORKFLOW_EXECUTION_LIFECYCLE_STAGE_ACCEPTED
+    );
+  }
+
+  /**
+   * Uses given input to make an UpdateWorkflowExecution call to the service, returning the update result.
+   *
+   * Used as the final function of the executeUpdate interceptor chain
+   */
+  protected async _executeUpdateHandler<Ret>(input: WorkflowUpdateInput): Promise<Ret> {
```

Comment:
You don't need that function, nor `_startUpdate`. Add `executionStage` to `WorkflowUpdateInput`, and then use only `_startUpdateHandler`.

Reply:
Also, these inner functions are about handling low level concerns; you should not return the `WorkflowUpdateHandle` object from here (ie. interceptors should never see the handle).

Reply:
Calling `handle.result()` should be in `WorkflowHandle.executeUpdate()`.